### ISSUE TITLE
Highlander causing reaper voicelines to not trigger

### ISSUE THREAD

**Issue by Iridar** [X2CommunityCore/X2WOTCCommunityHighlander#1419]
Source: https://www.reddit.com/r/xcom2mods/comments/1h3eitf/x2wotccommunityhighlander_causing_reaper/

> Wanted to return to modded xcom after a long break, had problems with reapers staying silent when doing blue move, certain abilities and on killing enemies.

> After some testing I found that highlander was the cause. Tried verifying files and resetting config but it only fixed the issue for blue move, reapers are still silent when killing aliens regardless of whether they had, had not or lost their shadow concealment after the killing shot.

**Comment by furudee** [comment]
Seems to happen with all hero classes
Debugger tells me it wants to play a personality line of whatever action, like Moving_INTENSE yet nothing happens after it's passed into PlayAkEvent in `XComCharacterVoice::PlaySoundForEvent`

Running console command like `MakeCharacterSpeak Moving_INTENSE` on a hero class produces nothing both in highlander and vanilla game, yet according to vanilla script file in `XGUnit::MaybeAddPersonalityToSpeech` it should be picking these same personality lines in vanilla game too, but it plays the lines normally

I made hero classes with base-game personalities so that it would pick a proper personality line according to vanilla script

I can't get DevSound log header to work even with `UnSuppress DevSound` so in vanilla game I don't know if it actually is trying to pass a personality voiceline into PlayAkEvent or not

**Comment by BlackDog86** [comment]
Yeah, I came to roughly the same conclusions. There are a few confounding factors though:

1. If the squad is concealed then it will use the MaybyUseHushedSpeechInstead function, which takes precedance over the personality lines so you need to make sure the squad is unconcealed before testing
2. If there is narrative playing, that also suppresses the voicelines. 

That said, your assessment seems more or less on point to me. Console commanding Moving_<PERSONALITY> doesn't work (for either normal or hero soldiers, base or CHL) so we can't test it that way.

However, I was able to reproduce the issue in the base game as well - When using a hero unit greater than veteran rank and unconcealed, it also tries to play one of the Moving_<PERSONALITY> lines and fails so I don't think this is purely a CHL issue, it's just relatively conditional.

I think if we know / assume that the hero classes don't have specific personality voicelines, we can just add the check for GameStateUnit.IsResistanceHero() to if (pidx!= INDEX_NONE) on MaybeAddPersonalityToSpeech in XGUnit. I've tested this in game and it works but I'm unsure about how it would interfere (if at all) with other mods adding attitudes. 

Should be notied that this mod already MCOs XGUnit which would make any changes in these functions redundant anyway:
https://steamcommunity.com/sharedfiles/filedetails/?id=1547952490

Still, we can't really tell what happens downstream of XComUnitPawnNativeBase UnitSpeak(nCharSpeech) and I couldn't find anything in the editor relating to the attitudes either.... 

To be contiuned?

**Comment by Dragon-32** [comment]
"TLP Attitude Voice Fix" was deprecated by Highlander 1.22.0, July 2021. So people running CHL shouldn't be using that mod at least.

**Comment by Iridar** [comment]
Fixed by #1422
